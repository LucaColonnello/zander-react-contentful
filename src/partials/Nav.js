import React from 'react'
import {NavLink, Link} from 'react-router-dom'

const Nav = (props) => {
  const navClass = `nav ${props.className}`;
  return (
    <nav className={navClass} role="navigation">
      <NavLink to="/articles" activeClassName="is-active" className="nav-link">Articles</NavLink>
      <NavLink to="/work" activeClassName="is-active" className="nav-link">Work</NavLink>
      {
        props.extras &&
          <NavLink to="/styleguide" activeClassName="is-active" title="View this site's style guide" className="nav-link">Style guide</NavLink>
      }
      {
        props.extras &&
          <Link to="http://github.com/mrmartineau/zander.wtf" title="Fork this site's source code" className="nav-link">Source</Link>
      }
    </nav>
  )
}

export default Nav
