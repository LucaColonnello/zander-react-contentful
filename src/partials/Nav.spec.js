import React from 'react'
import { shallow } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'
import Nav from './Nav'

test('Nav snapshot test', () => {
  const component = shallow(
    <Nav />
  )
  const tree = shallowToJson(component)
  expect(tree).toMatchSnapshot()
})

test('Nav with extras snapshot test', () => {
  const component = shallow(
    <Nav className="nav--footer" extras />
  )
  const tree = shallowToJson(component)
  expect(tree).toMatchSnapshot()
})
