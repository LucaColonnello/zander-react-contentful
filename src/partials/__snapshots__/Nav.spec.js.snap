// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav snapshot test 1`] = `
<nav
  className="nav undefined"
  role="navigation"
>
  <NavLink
    activeClassName="is-active"
    className="nav-link"
    to="/articles"
  >
    Articles
  </NavLink>
  <NavLink
    activeClassName="is-active"
    className="nav-link"
    to="/work"
  >
    Work
  </NavLink>
</nav>
`;

exports[`Nav with extras snapshot test 1`] = `
<nav
  className="nav nav--footer"
  role="navigation"
>
  <NavLink
    activeClassName="is-active"
    className="nav-link"
    to="/articles"
  >
    Articles
  </NavLink>
  <NavLink
    activeClassName="is-active"
    className="nav-link"
    to="/work"
  >
    Work
  </NavLink>
  <NavLink
    activeClassName="is-active"
    className="nav-link"
    title="View this site's style guide"
    to="/styleguide"
  >
    Style guide
  </NavLink>
  <Link
    className="nav-link"
    replace={false}
    title="Fork this site's source code"
    to="http://github.com/mrmartineau/zander.wtf"
  >
    Source
  </Link>
</nav>
`;
